prometheus/promhttp/instrument_server.go

```diff
@@ -322,23 +322,23 @@ func computeApproximateRequestSize(r *http.Request) int {
 func sanitizeMethod(m string) string {
 	switch m {
 	case "GET", "get":
-		return "get"
+		return "GET"
 	case "PUT", "put":
-		return "put"
+		return "PUT"
 	case "HEAD", "head":
-		return "head"
+		return "HEAD"
 	case "POST", "post":
-		return "post"
+		return "POST"
 	case "DELETE", "delete":
-		return "delete"
+		return "DELETE"
 	case "CONNECT", "connect":
-		return "connect"
+		return "CONNECT"
 	case "OPTIONS", "options":
-		return "options"
+		return "OPTIONS"
 	case "NOTIFY", "notify":
-		return "notify"
+		return "NOTIFY"
 	default:
-		return strings.ToLower(m)
+		return strings.ToUpper(m)
```

Comment:
The default rule will work for any case. No need to duplicate code. 
Or made the default case to error: non permitted HTTP method.

Reply:
> The default rule will work for any case. No need to duplicate code.

I don't understand. What code is duplicated here?

> Or made the default case to error: non permitted HTTP method.

The goal here is not to validate. If people really want to, they could come up with fantasy HTTP methods (or there might even be more added officially). The goal here is to make sure that the method is always upper case.

Reply:
If you remove all the cases, except the default one, the output will be the same. Or even more simple:
```
func sanitizeMethod(m string) string {
    return strings.ToUpper(m)
}
```

Reply:
Anyway, it's just a cosmetic suggestion for clarity, so just ignore it if it's not worth the time.